src/NUnitFramework/tests/Constraints/ThrowsExceptionConstraintTests.cs

```diff
@@ -65,19 +66,39 @@ public void SucceedsWithNonVoidReturningFunction()
         };
 
 #if ASYNC
+        private static async Task YieldAsync()
```

Comment:
I DRYed this out within this fixture, based on what I saw here: https://github.com/nunit/nunit/blob/cefc0c3a86c065daa4130aa255b1384db51b2b08/src/NUnitFramework/tests/Internal/TestExecutionContextTests.cs#L1002

Though this means we have 2 of these functions within the test suite. Is there a place I should move this?

Reply:
We could place it in the folder `nunit/src/NUnitFramework/tests/TestUtilities/`, but I don't think that there is a file that makes sense (`AsyncTestDelegates.cs` is probably the most relevant). And I think it would be overkill to create a new file just for this simple method. So I don't mind the minor duplication for now.

Reply:
What do you think of the name `YieldsAsync`?